Core/include/Acts/Utilities/Any.hpp

```diff
@@ -0,0 +1,495 @@
+  }
+
+  void moveConstruct(AnyBase&& fromAny) {
+    if (m_handler == nullptr) {
+      return;
+    }
+
+    void* to = dataPtr();
+    void* from = fromAny.dataPtr();
+    if (m_handler->heap) {
+      // stored on heap: just copy the pointer
+      setDataPtr(fromAny.dataPtr());
+      // do not delete in moved-from any
+      fromAny.m_handler = nullptr;
+      return;
+    }
+
+    if (m_handler->moveConstruct == nullptr) {
+      // trivially move constructible
+      m_data = std::move(fromAny.m_data);
+    } else {
+      m_handler->moveConstruct(from, to);
+    }
+  }
+
+  void move(AnyBase&& fromAny) {
+    if (m_handler == nullptr) {
+      return;
+    }
+
+    void* to = dataPtr();
+    void* from = fromAny.dataPtr();
+    if (m_handler->heap) {
+      // stored on heap: just copy the pointer
+      // need to delete existing pointer
+      m_handler->destroy(dataPtr());
+      setDataPtr(fromAny.dataPtr());
+      // do not delete in moved-from any
+      fromAny.m_handler = nullptr;
+      return;
+    }
+
+    if (m_handler->move == nullptr) {
+      // trivially movable
+      m_data = std::move(fromAny.m_data);
+    } else {
+      m_handler->move(from, to);
+    }
+  }
+
+  void copyConstruct(const AnyBase& fromAny) {
+    if (m_handler == nullptr) {
+      return;
+    }
+
+    void* to = dataPtr();
+    const void* from = fromAny.dataPtr();
+
+    if (m_handler->copyConstruct == nullptr) {
+      // trivially copy constructible
+      m_data = fromAny.m_data;
+    } else {
+      void* copyAt = m_handler->copyConstruct(from, to);
+      if (to == nullptr) {
+        assert(copyAt != nullptr);
+        // copy allocated, store pointer
+        setDataPtr(copyAt);
+      }
+    }
+  }
+
+  void copy(const AnyBase& fromAny) {
+    if (m_handler == nullptr) {
+      return;
+    }
+
+    void* to = dataPtr();
+    const void* from = fromAny.dataPtr();
+
+    if (m_handler->copy == nullptr) {
+      // trivially copyable
+      m_data = fromAny.m_data;
+    } else {
+      m_handler->copy(from, to);
+    }
+  }
+
+  template <typename T>
+  static void destroyImpl(void* ptr) {
+    assert(ptr != nullptr && "Address to destroy is nullptr");
+    T* obj = static_cast<T*>(ptr);
+    if constexpr (sizeof(T) <= SIZE) {
+      // stored in place: just call the destructor
+      _ACTS_ANY_VERBOSE("Destroy local at: " << ptr);
+      obj->~T();
+    } else {
+      // stored on heap: delete
+      _ACTS_ANY_DEBUG("Delete type: " << typeid(T).name()
+                                      << " heap at: " << obj);
+#if defined(_ACTS_ANY_ENABLE_TRACK_ALLOCATIONS)
+      {
+        std::lock_guard guard{_s_any_mutex};
+        auto it =
+            _s_any_allocations.find(std::pair{std::type_index(typeid(T)), obj});
+        if (it == _s_any_allocations.end()) {
+          throw std::runtime_error{
+              "Trying to deallocate heap address that we didn't allocate"};
+        }
+        _s_any_allocations.erase(it);
+      }
+#endif
```

Comment:
It is strange that your extracted `_ACTS_ANY_TRACK_ALLOCATION` above, but left this conditional code block inline, even though both are used only once now. Consider being consistent by either extracting both or neither (I'd be in favor of both, it would consolidate all the dirty ifdefs in a single place).

Reply:
It's only used once, that's why. But I moved it to a macro nonetheless.